lms/djangoapps/instructor/offline_gradecalc.py

```diff
@@ -1,117 +0,0 @@
-from django.contrib.auth.models import User
-from lms.djangoapps.grades import course_grades
-from opaque_keys import OpaqueKey
-from opaque_keys.edx.keys import UsageKey
-from xmodule.graders import Score
-
-from instructor.utils import DummyRequest
-
-
-class MyEncoder(JSONEncoder):
-    """ JSON Encoder that can encode OpaqueKeys """
-    def default(self, obj):  # pylint: disable=method-hidden
-        """ Encode an object that the default encoder hasn't been able to. """
-        if isinstance(obj, OpaqueKey):
-            return unicode(obj)
-        return JSONEncoder.default(self, obj)
-
-
-def offline_grade_calculation(course_key):
-    '''
-    Compute grades for all students for a specified course, and save results to the DB.
-    '''
-
-    tstart = time.time()
-    enrolled_students = User.objects.filter(
-        courseenrollment__course_id=course_key,
-        courseenrollment__is_active=1
-    ).prefetch_related("groups").order_by('username')
-
-    enc = MyEncoder()
-
-    print "{} enrolled students".format(len(enrolled_students))
-    course = get_course_by_id(course_key)
-
-    for student in enrolled_students:
-        request = DummyRequest()
-        request.user = student
-        request.session = {}
-
-        gradeset = course_grades.summary(student, course)
-        # Convert Score namedtuples to dicts:
-        totaled_scores = gradeset['totaled_scores']
-        for section in totaled_scores:
-            totaled_scores[section] = [score._asdict() for score in totaled_scores[section]]
-        gradeset['raw_scores'] = [score._asdict() for score in gradeset['raw_scores']]
-        # Encode as JSON and save:
-        gradeset_str = enc.encode(gradeset)
-        ocg, _created = models.OfflineComputedGrade.objects.get_or_create(user=student, course_id=course_key)
-        ocg.gradeset = gradeset_str
-        ocg.save()
-        print "%s done" % student  	# print statement used because this is run by a management command
-
-    tend = time.time()
-    dt = tend - tstart
-
-    ocgl = models.OfflineComputedGradeLog(course_id=course_key, seconds=dt, nstudents=len(enrolled_students))
-    ocgl.save()
-    print ocgl
-    print "All Done!"
-
-
-def offline_grades_available(course_key):
-    '''
-    Returns False if no offline grades available for specified course.
-    Otherwise returns latest log field entry about the available pre-computed grades.
-    '''
-    ocgl = models.OfflineComputedGradeLog.objects.filter(course_id=course_key)
-    if not ocgl:
-        return False
-    return ocgl.latest('created')
-
-
-def student_grades(student, request, course, use_offline=False):  # pylint: disable=unused-argument
-    '''
-    This is the main interface to get grades.  It has the same parameters as grades.grade, as well
-    as use_offline.  If use_offline is True then this will look for an offline computed gradeset in the DB.
-    '''
-    if not use_offline:
-        return course_grades.summary(student, course)
```

Comment:
Note the short-circuit logic here when using the default value of `use_offline`; this was refactored into the one active caller of this method.